### PR TITLE
DCOS-37617: Jobs / refactor: move logic from the mediator to graphql

### PR DESCRIPTION
**Pull request by Fabs**
Ressurect #3043 because CI wont build it anymore

### CODE REVIEW

**Comment by Fabs** [comment]
@juliangieseke. You are not able to stop more than one job by design. I did not dig deep to find the reason, but I believe it is due the way the api works, so if you like to stop 10 jobs, you would be issuing 10 request. Anyway, that was always the case before the Data-Layer.

Also, on #3043 there is a long argument on why I did not do the typeguards.

**Comment by juliangieseke** [comment]
I did not try to stop multiple jobs, just one… 🤔 

re Type Guards: I disagree on both of your arguments, it doesnt add a level of complexity - the opposite is true, it _reduces_ complexity (but increases _abstraction_). Also I dont think that `if (args.id)` is any better then `if (isSometype(args))`. 🤔

File: plugins/jobs/src/js/data/JobModel.ts [hunk 1]
```diff
@@ -141,7 +156,31 @@ export const resolvers = ({
       args: GeneralArgs,
       _context = {}
     ): Observable<JobLink> {
-      return runJob(args.id).map(({ jobId }) => ({ jobId }));
+      if (args.id) {
+        return runJob(args.id).map(({ jobId }) => ({ jobId }));
```
Comment (nLight) [inline plugins/jobs/src/js/data/JobModel.ts]: Generally I'd prefer an early return under test condition and not the main logic under a condition.

**Comment by nLight** [comment]
Confirmed, it's impossible to stop a job, the button does nothing

**Comment by nLight** [comment]
Works like a charm on master tho

**Comment by Fabs** [comment]
@juliangieseke. Ok will take a look on the stop not working.

**Comment by Fabs** [comment]
@nLight You are right, I reversed the other of the test to error in the oposite condition.

@juliangieseke left the if's instead of the type guards as we discussed.

**Comment by Fabs** [comment]
@nLight and @juliangieseke also the stop modal is working now.

**Comment by Fabs** [comment]
@juliangieseke there is a very old ticket linked to that, related to support stoping multiple jobs. And I also think that the fix here should come on the new jobs and queues. So I am hesitant to propose using radios as the solution here.

**Pull request merged**